### ISSUE TITLE
run_constrained is sometimes ignored

### ISSUE THREAD

**Issue by saraedum** [mamba-org/mamba#654]
prompt-toolkit and prompt_toolkit from conda-forge are connected by a `run_constrained`:

```
$ mamba create -n deleteme prompt_toolkit=1.0.15 prompt-toolkit=3.0.10
Problem: package prompt-toolkit-3.0.10-pyha770c72_0 has constraint prompt_toolkit 3.0.10 conflicting with prompt_toolkit-1.0.15-py27_0
```

However, doing the following we manage to break that constraint (and end up with a broken ipython/sagemath):

```
$ cat flatsurf/environment.yml
# This file lists the minimial dependencies needed to build and test
# sage-flatsurf. Create a conda environment with these dependencies
# preinstalled with:
# conda env create -f environment.minimal.yml
name: sage-flatsurf-build
channels:
  - flatsurf
  - conda-forge
  - defaults
dependencies:
  - gap-defaults
  - ipywidgets
  - matplotlib-base
  - pip
  - pytest
  - pytest-xdist
  - sagelib>=8.8
  - scipy
  - surface-dynamics
  - pip:
    - flipper
$ mamba env create -n deleteme -f flatsurf/environment.yml
$ mamba install -n deleteme sagelib=8.9
$ conda list -n deleteme | grep prompt
prompt-toolkit            3.0.10             pyha770c72_0    conda-forge
prompt_toolkit            1.0.15                     py_1    conda-forge
```

This happens with the latest version of mamba on conda-forge, see e.g. here: https://github.com/flatsurf/sage-flatsurf/pull/80/checks?check_run_id=1673467430.

When I replace mamba with conda in the `install` above, `prompt-toolkit` is removed, and `prompt_toolkit` is downgraded (which seems to be the right thing to do):
```
$ conda install -n deleteme sagelib=8.9
$ conda list -n deleteme | grep prompt
prompt_toolkit            1.0.15                     py_1    conda-forge
```

**Comment by wolfv** [comment]
Thanks for the bug report.

**Comment by wolfv** [comment]
fixed with #773
